### PR TITLE
Pivotal ID # 187898943: Submission Files Integrity Check

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/187898943

- Check integrity of files and directories persisted on NFS
- Check integrity of generated zip files
- Add tests
- Remove unused code

### CODE REVIEW

File: commons/commons-util/src/main/kotlin/ebi/ac/uk/io/FileUtils.kt [hunk 1]
```diff
@@ -104,6 +106,18 @@ object FileUtils {
         }
     }
 
+    fun checkFileIntegrity(target: File, expectedMd5: String) {
+        require(target.md5() == expectedMd5) { throw CorruptedFileException(target) }
+    }
+
+    fun checkDirectoryIntegrity(
```
Comment (Juan-EBI) [inline commons/commons-util/src/main/kotlin/ebi/ac/uk/io/FileUtils.kt:113]: same as above

File: submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/ZipUtil.kt [hunk 2]
```diff
@@ -39,4 +46,20 @@ object ZipUtil {
...
+            listAllFilesWithRelPaths(sourceDir).forEach {
+                val (path, sourceFile) = it
+                val entry = zip.getEntry(path)
+
+                if (entry == null || sourceFile.md5() != zip.md5(entry)) throw CorruptedFileException(sourceFile)
+            }
+        }
+    }
+
+    private fun ZipFile.md5(entry: ZipEntry): String = getInputStream(entry).use { DigestUtils.md5Hex(it).uppercase() }
```
Comment (Juan-EBI) [inline submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/ZipUtil.kt:64]: with this change we are calculating the md5 twice, this is impacting significantly the zip time for large folder. It looks like a bad idea IMO, still if that is what need it well we can do it but IMO this is doing more harm than good to the system. Ugis will have lass optinion

File: submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/ZipUtil.kt [hunk 3]
```diff
@@ -22,6 +27,8 @@ object ZipUtil {
                     zs.closeEntry()
                 }
         }
+
+        checkZipIntegrity(sourceDir, zipFile)
```
Comment (Juan-EBI) [inline submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/ZipUtil.kt:31]: this is a generic zip util I think we should not include the double file check as part of the utility but rather do it in biostudies business code.

**Review by Juan-EBI — COMMENTED** [review]
Looking good please check comments

**Comment by jhoanmanuelms** [comment]
We should come back to this PR when we actually start working on the ticket

**Pull request closed without merging**